### PR TITLE
AMLII-1110 - Fixed UseSSL setting for HTTP endpoints and addiitional endpoints

### PR DESCRIPTION
**Pull request by soberpeach**
### What does this PR do?
This is a set of two related fixes regarding the `UseSSL` setting for agent logs endpoints.

The first fix changes the behavior of the additional logs endpoints (under the logs config in the agent) to respect their respective `logs_no_ssl` config setting instead of automatically inheriting from the main endpoint.

The second fix is for HTTP endpoints to respect the `logs_no_ssl` config option instead of automatically overriding it depending on the `dd_url` config option.

### Motivation

These features have been broken since inception, this change fixes them.

### Possible Drawbacks / Trade-offs

This is a potentially breaking change to some workflows. Previously if the `dd_url` option was set to a URL with an `https://` prefix, the agent would have used SSL to send logs, regardless of what the `logs_no_ssl` option was set to. This fixes it so the `logs_no_ssl` option takes precedence.

### Describe how to test/QA your changes

Enable logging in the agent and run it with additional additional logs endpoints (example provided below). Check if the agent is sending logs to the additional endpoint using SSL or not using SSL according to the configuration using `agent status -v`. The additional endpoints should appear below the main endpoint in the output of the logs agent.

Sample additional endpoint configuration:
```
logs_config:
  use_http: true
  additional_endpoints:
  - api_key: "apiKey2"
    Host: "randval"
    Port: 443
    is_reliable: true
    logs_no_ssl: false
```

For the main endpoint `logs_no_ssl` setting, it is possible to use various combinations of `http` and `https` endpoints and `logs_no_ssl` settings to make sure the resulting endpoints use the correct settings. The truth table should look as follows:

`dd_url` Prefix | `logs_no_ssl` | endpoint result
------|--------|------------
HTTPS  | true  | HTTP
HTTPS  | false | HTTPS
HTTP   | true  | HTTP
HTTP   | false | HTTP

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: comp/logs/agent/config/config_test.go [hunk 1]
```diff
@@ -361,7 +369,7 @@ func (suite *ConfigTestSuite) TestMultipleHttpEndpointsInConfig2() {
 		APIKey:           "456",
 		Host:             "additional.endpoint.1",
 		Port:             1234,
-		UseSSL:           true,
+		UseSSL:           &trueVal,
```
Comment (gh123man) [inline comp/logs/agent/config/config_test.go:372]: Absolutely nothing wrong with this, but we also have a [helper for generating pointers](https://github.com/DataDog/datadog-agent/blob/c89808ba9d8c593d5fabe672ee00220418bb5060/pkg/util/pointer/pointer.go#L10) for tests like this:
`pointer.Ptr(true)`

File: comp/logs/agent/config/endpoints.go [hunk 2]
```diff
@@ -68,7 +68,7 @@ func (e *Endpoint) GetStatus(prefix string, useHTTP bool) string {
 
 	var protocol string
 	if useHTTP {
-		if e.UseSSL {
+		if e.UseSSL == nil || *e.UseSSL {
```
Comment (gh123man) [inline comp/logs/agent/config/endpoints.go:71]: Maybe we can make a helper function on `Endpoint` that returns `e.UseSSL == nil || *e.UseSSL` so you don't have to duplicate this code in several places

File: comp/logs/agent/config/config_test.go (hunk 1 above)
Comment (soberpeach) [reply comp/logs/agent/config/config_test.go:372]: Ooo didn't know about that, I'll use that instead

File: comp/logs/agent/config/endpoints.go (hunk 2 above)
Comment (soberpeach) [reply comp/logs/agent/config/endpoints.go:71]: Good idea

**Pull request closed without merging**